Automattic/wp-calypso
# Site Pages: implement the popup menu using EllipsisMenu and Redux

This PR started out as an attempt to remove usages of `lib/posts/actions` (the post list Flux store actions) from the Site Pages list. The Flux store is no longer used by anyone to read from (it's all Redux now) and the Flux actions only send the REST API requests and the update an unused Flux store. Redux is not updated at all.

The Site Pages list is the only remaining usage of the `updatePostStatus` HoC that adds capability to display UI like this to the `Page` component:

![page-update-post-status](https://user-images.githubusercontent.com/664258/37109432-c6531a9a-21ee-11e8-8782-afa89a074d2b.gif)

Here I do the following:
- start dispatching Redux actions instead of Flux ones
- remove the overlay UI from `updatePostStatus`: after migrating to Redux, notices are shown when a post status is succesfully changed
- remove the capability of `updatePostStatus` to maintain opened/closed state of the popover menu and start using `EllipsisMenu` for that
- merge the `updatePostStatus` HoC into the `Page` component. At this moment, only the `updatePostStatus` method remains

The result is that the Site Pages now behave exactly the same as Blog Posts. The post list was redesigned by @Automattic/tanooki (master thread is at p8F9tW-vh-p2) recently and Site Pages were out of scope of that project. Now, the Site Pages list still doesn't use `PostTypeList`, but the UI behavior is very close.

**How to test:**
1. Go to a site with a few pages
2. Open the popup menu for a page and click "Trash". Verify that the page was moved to trash and that a success notice is shown.
4. Open the list of trashed pages and try to restore one. Verify that the trashed page was moved to "Drafts" (the 100% right behavior would be to move it to "Published" -- that's a pre-existing bug I'd like to fix soon)
5. In the list of trashed pages, try to delete one permanently ("Delete" action). Verify that the permanent delete really happened.
6. In the list of "Drafts", click on the "Publish" action in the menu. Verify that the page was moved to "Published"
7. Verify that the UI behavior is the same as in Blog Posts page and there are no significant differences.
